recommenders/models/sasrec/sampler.py

```diff
@@ -0,0 +1,86 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+# Original codes are from
+# https://github.com/kang205/SASRec/blob/master/sampler.py
+import numpy as np
+from multiprocessing import Process, Queue
+
+
+def random_neq(left, right, s):
+    t = np.random.randint(left, right)
+    while t in s:
+        t = np.random.randint(left, right)
+    return t
+
+
+def sample_function(
+    user_train, usernum, itemnum, batch_size, maxlen, result_queue, seed
+):
+    """
+    Batch sampler that creates a sequence of negative items based on the
+    original sequence of items (positive) that the user has interacted with.
+    """
+
```

Comment:
if you have time, it would be great to add the missing docstrings. This is an example of the format we follow:
```
    def check_column_dtypes_wrapper(
        rating_true,
        rating_pred,
        col_user=DEFAULT_USER_COL,
        col_item=DEFAULT_ITEM_COL,
        col_rating=DEFAULT_RATING_COL,
        col_prediction=DEFAULT_PREDICTION_COL,
        *args,
        **kwargs
    ):
        """Check columns of DataFrame inputs
        Args:
            rating_true (pandas.DataFrame): True data
            rating_pred (pandas.DataFrame): Predicted data
            col_user (str): column name for user
            col_item (str): column name for item
            col_rating (str): column name for rating
            col_prediction (str): column name for prediction
        """
```
see here more info: https://github.com/microsoft/recommenders/blob/main/recommenders/evaluation/python_evaluation.py#L51